src/psyclone/tests/dynamo0p3_lma_test.py

```diff
@@ -101,51 +101,85 @@ def test_get_op_wrong_name():
     assert "Unsupported name 'not_an_op' found" in str(err.value)
 
 
+def test_ad_op_type_invalid_data_type():
+    ''' Tests that an error is raised when the argument descriptor
+    metadata for an operator has an invalid data type. '''
+    code = CODE.replace(
+        "arg_type(gh_operator, gh_real,    gh_read, w2, w2)",
+        "arg_type(gh_operator, gh_clear,    gh_read, w2)", 1)
+    ast = fpapi.parse(code, ignore_comments=False)
+    name = "testkern_qr_type"
+    with pytest.raises(ParseError) as excinfo:
+        _ = DynKernMetadata(ast, name=name)
+    assert ("In the LFRic API the 2nd argument of a 'meta_arg' entry should "
+            "be a valid data type (one of ['gh_real', 'gh_integer']), but "
+            "found 'gh_clear' in 'arg_type(gh_operator, gh_clear, gh_read, "
+            "w2)'." in str(excinfo.value))
+
+
 def test_ad_op_type_too_few_args():
     ''' Tests that an error is raised when the operator descriptor
-    metadata has fewer than 4 args. '''
-    code = CODE.replace("arg_type(gh_operator, gh_read, w2, w2)",
-                        "arg_type(gh_operator, gh_read, w2)", 1)
+    metadata has fewer than 5 args. '''
+    code = CODE.replace(
+        "arg_type(gh_operator, gh_real,    gh_read, w2, w2)",
+        "arg_type(gh_operator, gh_real,    gh_read, w2)", 1)
     ast = fpapi.parse(code, ignore_comments=False)
     name = "testkern_qr_type"
     with pytest.raises(ParseError) as excinfo:
         _ = DynKernMetadata(ast, name=name)
-    assert ("'meta_arg' entry must have 4 arguments if its first "
+    assert ("'meta_arg' entry must have 5 arguments if its first "
             "argument is an operator (one of {0})".
             format(LFRicArgDescriptor.VALID_OPERATOR_NAMES) in
             str(excinfo.value))
 
 
 def test_ad_op_type_too_many_args():
     ''' Tests that an error is raised when the operator descriptor
-    metadata has more than 4 args. '''
-    code = CODE.replace("arg_type(gh_operator, gh_read, w2, w2)",
-                        "arg_type(gh_operator, gh_read, w2, w2, w2)", 1)
+    metadata has more than 5 args. '''
+    code = CODE.replace(
+        "arg_type(gh_operator, gh_real,    gh_read, w2, w2)",
+        "arg_type(gh_operator, gh_real,    gh_read, w2, w2, w2)", 1)
+    ast = fpapi.parse(code, ignore_comments=False)
+    name = "testkern_qr_type"
+    with pytest.raises(ParseError) as excinfo:
+        _ = DynKernMetadata(ast, name=name)
+    assert "'meta_arg' entry must have 5 arguments" in str(excinfo.value)
+
+
+def test_ad_op_type_invalid_fs_to():
```

Comment:
Is this a duplicate of `test_ad_op_type_1st_arg_not_space`?

Reply:
I just realised it is. I tried to better organise the tests. `def test_arg_descriptor_invalid_fs2` from `dynamo0p3_test.py` became the `test_ad_op_type_invalid_fs_from` here so the function spaces to- and from- for operators are not split over two files. I think that the name `test_ad_op_type_1st_arg_not_space` is confusing as it tests for the function space to- which is the 4th argument in the metadata. I removed `test_ad_op_type_invalid_fs_to` and renamed `test_ad_op_type_1st_arg_not_space` to `test_ad_op_type_4th_arg_not_space`. I also renamed `test_ad_op_type_invalid_fs_from` to  `test_ad_op_type_5th_arg_not_space` for consistency.

Reply:
Thanks Iva.